dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.hdlresolver;
+
+import java.text.MessageFormat;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.handle.hdlresolver.HdlResolverDTO;
+import org.dspace.handle.hdlresolver.HdlResolverService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller is public and is useful for handle resolving,
+ * wheter a target handle identifier will be resolved into the
+ * corresponding URL (if found), otherwise will respond a null string.
+ * 
+ * @author Vincenzo Mecca (vins01-4science - vincenzo.mecca at 4science.it)
+ *
+ */
+@RestController
```

Comment:
I would recommend we offer a way to disable this controller, as it seems like it's only needed for sites that use https://github.com/DSpace/Remote-Handle-Resolver. For any other repositories it's unnecessary and could bog down the site if crawlers hit it excessively.

Perhaps we could add something like:
```
@ConditionalOnProperty("handle.remote-resolver.enabled")
```

And then add a new `handle.remote-resolver.enabled = false` (by default) to the dspace.cfg alongside the other `handle.*` settings in that file?
```
# Whether to enable the DSpace handle resolver endpoints necessary for
# https://github.com/DSpace/Remote-Handle-Resolver
# Defaults to "false" which means these handle resolver endpoints are not available.
#handle.remote-resolver.enabled = false
```

I noticed we have an older configuration called `handle.hide.listhandles` which used to serve a similar purpose. We can remove that unused config and replace it with this new one.

Reply:
Done, replaced with the new property and used to conditionally create the `HdlResolver`.